packages/core/test/crypto.spec.ts

```diff
@@ -136,98 +132,51 @@ describe("Crypto", () => {
-      const topic = utils.generateRandomBytes32();
-      const result = crypto.encode(topic, payload);
-      expect(result).to.equal(encoding.utf8ToHex(safeJsonStringify(payload)));
+      await expect(invalidCrypto.encode("topic", payload)).to.eventually.be.rejectedWith(
+        "Not initialized. crypto",
+      );
     });
     it("encrypts `payload` if the passed topic is known", async () => {
-      const symKey = utils.generateRandomBytes32();
       const topic = await crypto.setSymKey(symKey);
-      const spy = Sinon.spy();
-      crypto.encrypt = spy;
-      crypto.encode(topic, payload);
-      const [calledTopic, calledMessage] = spy.getCall(0).args;
-      expect(calledTopic).to.equal(topic);
-      expect(calledMessage).to.equal(safeJsonStringify(payload));
+      // const spy = Sinon.spy();
+      // crypto.encrypt = spy;
+      // FIXME: needs to be tested dynamically because of random IV generation
+      await crypto.encode(topic, payload);
+      // const [calledTopic, calledMessage] = spy.getCall(0).args;
+      // expect(calledTopic).to.equal(topic);
+      // expect(calledMessage).to.equal(safeJsonStringify(payload));
     });
   });
 
   describe("decode", () => {
+    const symKey = "5720435e682cd03ee45b484f9a213f0e3246a0ccc2cca183b72ab1cbfbefb702";
     const payload = { id: 1, jsonrpc: "2.0", result: "result" };
-    const hexPayload = encoding.utf8ToHex(safeJsonStringify(payload));
+    const encoded =
+      "AG7iJl9mMl9K04REnuWaKLQU6kwMcQWUd69OxGOJ5/A+VRRKkxnKhBeIAl4JRaIft3qZKEfnBvc7/Fife1DWcERqAfJwzPI=";
 
-    it("throws if not initialized", () => {
+    it("throws if not initialized", async () => {
       const invalidCrypto = new Crypto(core, logger);
-      expect(() => invalidCrypto.decode("topic", "encoded")).to.throw("Not initialized. crypto");
-    });
-    it("decodes `encoded` from hex to utf8 string if the passed topic is not known", () => {
-      const topic = utils.generateRandomBytes32();
-      const result = crypto.decode(topic, hexPayload);
-      expect(result).to.deep.equal(safeJsonParse(encoding.hexToUtf8(hexPayload)));
+      await expect(invalidCrypto.decode("topic", "encoded")).to.eventually.be.rejectedWith(
+        "Not initialized. crypto",
+      );
     });
     it("decrypts `payload` if the passed topic is known", async () => {
-      const symKey = utils.generateRandomBytes32();
       const topic = await crypto.setSymKey(symKey);
-      const spy = Sinon.spy(() => "message");
-      crypto.decrypt = spy;
-      crypto.decode(topic, hexPayload);
-      const [calledTopic, calledEncoded] = spy.getCall(0).args;
-      expect(calledTopic).to.equal(topic);
-      expect(calledEncoded).to.equal(hexPayload);
+      // const spy = Sinon.spy(() => "message");
+      // crypto.decrypt = spy;
+      const decoded = await crypto.decode(topic, encoded);
+      expect(decoded).to.eql(payload);
+      // const [calledTopic, calledEncoded] = spy.getCall(0).args;
```

Comment:
I commented out because I wasn't sure if you felt it would be necessary or not

In this case Im going to guess its not longer necessary so I will remove them